content/en/apps/tutorials/application-tests.md

```diff
@@ -0,0 +1,534 @@
+
+
+## Writing a test for CHT App
+
+Start by adding a file where a group of related tests are written. For example, in the default config, all tests for the pregnancy form are in the [pregnancy.spec.js](https://github.com/medic/cht-core/blob/master/config/default/test/forms/pregnancy.spec.js) file.
+
+Let's introduce some important sections from a typical test file:
+
+
+```js
+const TestRunner = require('cht-conf-test-harness');
+const harness = new TestRunner();
+```
+
+
+You get an instance of `harness` and the same instance is used throughout the test file.
+
+You can also pass parameters to the `TestRunner()` when instantiating:
+```js
+// For detailed console logs
+const harness = new TestRunner({ verbose: true });
+```
+
+Other useful options are:
+1. `{ headless: false }` -  Passed to [puppeteer](https://developers.google.com/web/tools/puppeteer/get-started#default_runtime_settings), launches browser with GUI. Helpful to see how the form is getting filled.
+2. `{ harnessDataPath: 'harness.clinic.json'}`: Specify different harness confguration
+
+You can find more harness options and examples [here](https://docs.communityhealthtoolkit.org/cht-conf-test-harness/Harness.html).
+
+
+You need to start the harness before running tests and stop it after the use. To handle start and stop of harness, place following statements in `before()` and `after()` hooks of the test suite.
+```js
+describe('PNC form tests', () => {
+  before(async () => { return await harness.start(); });
+  after(async () => { return await harness.stop(); });
+```
+
+Any logic that you want to execute before and after each test can be placed in the `beforeEach()` and `afterEach()` hooks. Clear the state of the harness and set the date to a fixed value before each test run:
+```js
+  beforeEach(
+    async () => {
+      await harness.clear();
+      await harness.setNow('2000-01-01');
+    });
+```
+
+After each test run, assert that there are no errors in the console.
+```js
+  afterEach(() => {
+    expect(harness.consoleErrors).to.be.empty;
+  });
+```
+
+If you want to learn more about these hooks, refer to this [Mocha resource](https://mochajs.org/#hooks). Feel free to customize the hooks as you see fit.
+
+Let's look at a more detailed example. <a name="assessment-form-test">Here</a> is a test case for the Assessment form that was covered in the [previous tutorial]({{< ref "apps/tutorials/app-forms" >}}):
+
+{{< highlight js "linenos=table,hl_lines=26 31 36,anchorlinenos=true,lineanchors=assessment-form-test" >}}
+  it('unit test confirming assessment with cough since 7 days', async () => {
+    // Load the assessment form and fill 'yes' on the first page and '7' on the second page
```

Comment:
Can you please split this comment into two lines? Something like below. I just noticed that it exceeds the code block box unlike other code snippets.

    // Load the assessment form and fill 'yes' on the first page 
    // and '7' on the second page

Reply:
Done, thanks. :)